### PR TITLE
Added User profile page

### PR DESCRIPTION
**Pull request by MySmilingTurban**
This PR adds content for Profile page (screenshots attached)
Profile pictures are stored in Firebase Cloud Storage and its URL is added to user's photoURL property

Screenshot 2024-01-21 at 3 22 04 pm

User can view their details in the above format

Screenshot 2024-01-21 at 3 22 30 pm

Name, username and profile picture can be edited

Screenshot 2024-01-21 at 3 24 39 pm

Notifies user if username is already in use (as usernames are supposed to be unique)

### CODE REVIEW

**Comment by amitojsingh** [comment]
@MySmilingTurban veer ji can we add a thumbnail for profile pic when we upload? Also can we move all the content to center with a transparent box as our web app is more centric. Also can we make button (Verify/save/upload/choose file) design similar to start button just for the consistency.

**Comment by MySmilingTurban** [comment]
Fixed as per suggestions

Screenshot 2024-01-23 at 11 39 52 am

Screenshot 2024-01-23 at 11 40 12 am


https://github.com/KhalisFoundation/gurmukhi-web/assets/51584037/e4ca3a1d-a4a8-4307-a6bf-b7bac12a3e47

cc: @amitojsingh

**Comment by amitojsingh** [comment]
@MySmilingTurban can you check why deploy is failing?

**Comment by MySmilingTurban** [comment]
> @MySmilingTurban can you check why deploy is failing?

fixed

File: src/pages/dashboard/components/wordBox/index.tsx [hunk 1]
```diff
@@ -17,7 +19,9 @@ function WordBox({ commonStyle }: { commonStyle: string }) {
       setRandomWord(newWord);
     };
 
-    fetchData();
+    if (user && user.uid) {
+      fetchData();
+    }
   }, []);
```
Comment (amitojsingh) [inline src/pages/dashboard/components/wordBox/index.tsx:25]: please add user in the useEffect dependency

**Review by amitojsingh — CHANGES_REQUESTED** [review]

File: src/pages/dashboard/components/wordBox/index.tsx (hunk 1 above)
Comment (MySmilingTurban) [reply src/pages/dashboard/components/wordBox/index.tsx]: fixed

**Comment by amitojsingh** [comment]
@MySmilingTurban veer ji can you add a full page loading or some animation? I can see a small loader once we save. Also, make it reusable so that we can use it in our entire app where ever loading is required.

**Review by amitojsingh — APPROVED** [review]

**Comment by amitojsingh** [comment]
For Loading we can create new task. Let's resolve conflicts and merge. 
Looks good to me. Great Work Veer ji. 
@MySmilingTurban

**Pull request merged**